packages/core/src/messageformat.ts

```diff
@@ -25,12 +25,37 @@ export type MessageFunction<ReturnType extends 'string' | 'values'> = (
  *
  * @public
  */
-export type CustomFormatter = (
+export type DefaultCustomFormatter = (
   value: unknown,
   locale: string,
   arg: string | null
 ) => unknown;
 
+/**
+ * A custom formatter function that is bound to specific locale formatting
+ * module. This formatting signature is used when you specify a `module`
+ * thunk, generating a module based on `localeCode`.
+ * {@link https://messageformat.github.io/messageformat/custom-formatters/ | Custom Formatters}
+ * for more details.
+ *
+ * @public
+ */
+export type LocaleModuleCustomFormatter = (
+  value: unknown,
+  arg: string | null
+) => unknown;
+
+/**
+ * A custom formatter function. See
+ * {@link https://messageformat.github.io/messageformat/custom-formatters/ | Custom Formatters}
+ * for more details.
+ *
+ * @public
+ */
+export type CustomFormatter =
+  | DefaultCustomFormatter
+  | LocaleModuleCustomFormatter;
```

Comment:
As we're now always including the locale code, am I right that the changes to this signature should be rolled back?

Reply:
No, otherwise I’m in no better position than when I started without the PR. the function form is precisely what lets me generate locale based module imports. The param is runtime, versus compile time

Reply:
The function form of `module` allows that. This is the signature of the function that'll be called during the formatting. After https://github.com/messageformat/messageformat/pull/403#discussion_r1272317743, it'll always include the locale argument.

Reply:
got it. i misinterpreted. my bad. addressed in a681a58